ledger/eval.go

```diff
@@ -768,20 +763,6 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 		return fmt.Errorf("transaction %v: %v", txid, err)
 	}
 
-	// Validate applyData if we are validating an existing block.
```

Comment:
if we do not check entire apply data, then we need to check maybe a hash of it? Can't find it.

Reply:
yes, I believe you'll find it in finalValidation : we compute the commit hash for the block.
The flat commit hash is computed as the hash of the entire payset ( with apply data ).
so, by verifying the root hash, we can avoid comparing all the inner transactions.

Reply:
There is a code path to avoid this validation: `TransactionPool.addToPendingBlockEvaluatorOnce` calls `BlockEvaluator.TransactionGroup` directly so `finalValidation` is not called.

Reply:
`finalValidation` is called by `eval` which is called by `Validate`, am I missing something?

Reply:
(all in `ledger/eval.go`)

Reply:
The codepath through `TransactionPool.addToPendingBlockEvaluatorOnce` ( which truly doesn't have the `finalValidation` ), is always providing dummy apply data anyway, so I don't think it's meaningful anyway.
( i.e. `addToPendingBlockEvaluatorOnce` converts a `[]transactions.SignedTxn` into `[]transactions.SignedTxnWithAD` with a simple copy.. )

Reply:
Then I'm good